src/Components/IndexScene/IndexScene.tsx

```diff
@@ -221,6 +222,33 @@ export class IndexScene extends SceneObjectBase<IndexSceneState> {
     });
   }
 
+  private subscribeToPasteTimeEvent = async () => {
+    const copiedRange = await getCopiedTimeRange();
+
+    if (!copiedRange.isError) {
+      const timeRange = sceneGraph.getTimeRange(this);
+      const to = typeof copiedRange.range.to === 'string' ? copiedRange.range.to : undefined;
+      const from = typeof copiedRange.range.from === 'string' ? copiedRange.range.from : undefined;
+
+      const newRange = rangeUtil.convertRawToRange(copiedRange.range);
+
+      if (timeRange && newRange) {
+        timeRange.setState({
+          value: newRange,
+          to,
+          from,
+        });
+      } else {
+        logger.error(new Error('Invalid time range from clipboard'), {
+          msg: 'Invalid time range from clipboard',
+          sceneTimeRange: typeof timeRange,
+          to: to ?? '',
+          from: from ?? '',
+        });
+      }
+    }
+  };
```

Comment:
As usual, one less level of indentation.

```suggestion
  private subscribeToPasteTimeEvent = async () => {
    const copiedRange = await getCopiedTimeRange();

    if (!copiedRange.isError) {
      return;
    }
    const timeRange = sceneGraph.getTimeRange(this);
    const to = typeof copiedRange.range.to === 'string' ? copiedRange.range.to : undefined;
    const from = typeof copiedRange.range.from === 'string' ? copiedRange.range.from : undefined;

    const newRange = rangeUtil.convertRawToRange(copiedRange.range);

    if (timeRange && newRange) {
      timeRange.setState({
        value: newRange,
        to,
        from,
      });
    } else {
      logger.error(new Error('Invalid time range from clipboard'), {
        msg: 'Invalid time range from clipboard',
        sceneTimeRange: typeof timeRange,
        to: to ?? '',
        from: from ?? '',
      });
    }
  };
```